cluster/saltbase/salt/kube-addons/kube-addons.sh

```diff
@@ -134,6 +165,22 @@ for k,v in yaml.load(sys.stdin).iteritems():
 ''' < "${kube_env_yaml}")
 fi
 
+# Delete old default namespaced resources.  Necessary for update
+# TODO Needs testing
+#for name in elasticsearch-logging-v1 kibana-logging-v1 kube-dns-v3 monitoring-heapster-v4 monitoring-influx-grafana-v1; do
```

Comment:
better to delete all objects with kubernetes.io/cluster-service=true. No matter what versions they are. In this way, an update from a slightly older version will also work.